Documentation/contributors/design-proposals/webhooks.md

```diff
@@ -0,0 +1,213 @@
+          clientConfig:
+            service:
+              name: mywebhook-service
+              namespace: test-ns
+              path: "/admit"
+
+Note that the CABundle field in the clientConfig is set to use the general CA from the Kubernetes apiserver.
+
+### Description for v2
+
+The following example is a different take which would remove control from the user and have OLM manage the missing bits that aren't specified. The missing fields and objects would be created by OLM on the fly.
+
+### CSV example that includes webhook configuration (v2)
+
+This example serves as a potential method of CSV configuration that gives minimal control to the user. Portions of configuration that have been eliminated are commented out in order to make an easier comparison.
+
+  install:
+    strategy: deployment
+    spec:
+      clusterPermissions:
+        - serviceAccountName: my-operator
+          rules:
+          - apiGroups:
+            - "*"
+            ...
+      permissions:
+        - serviceAccountName: my-operator
+          rules:
+          - apiGroups:
+            - ""
+            ...
+      deployments:
+        - name: webhook-deployment
+          spec:
+            replicas: 1
+            selector:
+              matchLabels:
+                name: mywebhook
+            template:
+              metadata:
+                labels:
+                  name: mywebhook
+              spec:
+                serviceAccountName: my-operator
+                containers:
+                - name: webhook
+                  image: operator-framework/fake-webhook-image:latest
+                  imagePullPolicy: IfNotPresent
+                  ports:
+                  - containerPort: 443
+                  env:
+                  - name: NAMESPACE
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.annotations['olm.targetNamespaces']
+                  #volumeMounts:
+                  #  - mountPath: /keys
+                  #    name: tls-keys
+                  #    readonly: true
+                - name: operator
+                  image: operator-framework/fake-operator:latest
+                  imagePullPolicy: IfNotPresent
+                  env:
+                  - name: NAMESPACE
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.annotations['olm.targetNamespaces']
+                  - name: MY_POD_NAME
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.name
+                #volumes:
+                #- name: tls-keys
+                #    secretName: service-tls
+      #services:
+      #- name: mywebhook-service
+      #  namespace: test-ns
+      #  spec:
+      #    selector:
+      #      name: mywebhook
+      #    ports:
+      #    - protocol: TCP
+      #      port 443
+      #      targetPort: 443
+      validatingWebhooks:
```

Comment:
Pros/cons of something like this?
```yaml
webhooks:
- name: olm-admission
   type: ValidatingAdmissionWebhook
   spec:
- name: olm-mutate
   type: MutatingAdmissionWebhook
```
?

Reply:
The only drawback to this approach is if you had several webhooks of the same type, the previous format would allow you to add them as another list item rather than creating a new section in webhooks as your example above. Given though that it's probably rare that multiple webhooks of the same type are present, I'm going to reframe the proposal using what is proposed here.